content/en/blog/2020/custom-istio-dns-name/index.md

```diff
@@ -0,0 +1,125 @@
+
+    {{< text bash >}}
+    $ kubectl apply --context=$CTX_CLUSTER1 -n foo -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: httpbin-bar
+    spec:
+      hosts:
+      # must be of form name.namespace.global
+      - httpbin.bar.global
+      # Treat remote cluster services as part of the service mesh
+      # as all clusters in the service mesh share the same root of trust.
+      location: MESH_INTERNAL
+      ports:
+      - name: http1
+        number: 8000
+        protocol: http
+      resolution: DNS
+      addresses:
+      - 240.0.0.2
+      endpoints:
+      - address: ${CLUSTER2_GW_ADDR}
+        ports:
+          http1: 15443 # Do not change this port value
+    EOF
+    {{< /text >}}
+
+This `ServiceEntry` creates an entry in the Istio service-discovery registry for `httpbin.bar.global` with endpoints that point to a different cluster's gateway. Without any other configuration changes, this hostname, `httpbin.bar.global` is not routable. In the docs for setting up multi-cluster replicated control plane, however, there is a [section for configuring the DNS stubbing](/docs/setup/install/multicluster/gateways/#setup-dns) for your Kubernetes cluster. When we set up the DNS stubbing, we tell `kube-dns` to delegate queries for `.global` to the Istio coredns component.
+
+But what if you don't want to use `.global` as the suffix? What if you have to keep some backward compatibility and the root domain name needs to be something custom? Or what if you want to be able to support multiple root domain suffixes? In the open-source project [Service Mesh Hub](https://github.com/solo-io/service-mesh-hub) that I work on, we had this exact scenario. Service Mesh Hub is used to [automate multi-cluster Istio service mesh deployments](https://docs.solo.io/service-mesh-hub/latest/getting_started/), and routing between clusters is a use case our users have. Here's how to configure it in Istio.
+
+## Istio CoreDNS
+
+Istio's operator allows you to deploy the Istio `coredns` addon component.  For example, you can specify the following in the operator config:
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: istiocontrolplane-default
+  namespace: istio-operator
+spec:
+  addonComponents:
+    istiocoredns:
+      enabled: true
+{{< /text >}}
+
+When this deploys, we have the `istiocoredns` component that has `coredns` installed as well as the `istio-coredns-plugin` which is used to read `ServiceEntry`s and serve DNS queries for their hostname with IP addresses. The interesting bit is how it's configured out of the box. The default out of the box config is in the `coredns` configmap:
```

Comment:
Probably best to add a noun after "this" to clarify exactly what it is that's deploying. (https://developers.google.com/style/pronouns#ambiguous-pronoun-references)